### PR TITLE
wire pstake liquid-staking cosmos module.

### PR DESCRIPTION
**Pull request by puneet2019**
## 1. Overview

wires 
- persistence-sdk/epochs
- persistence-sdk/ibchooker
- persistence-sdk/interchainquery
- pstake-native/lscosmos
- persistence-sdk/halving
 
## 2. Implementation details

- app wiring
- ibchooker is wrapper around ibc-transfer
- lscosmos is auth module for ica controller
- lscosmos implements epoch hooks, ibchooker hooks.
- lscosmos implements interchainquery callbacks
- removes ./x/halving and adds persistence-sdk/halving
- removes thirdparty protos and compiling scripts

## 3. How to test/use
- do txns of the modules
- test upgrade

## 4. Checklist
- [ ] test Upgrade works
- [ ] All transactions work and are as expected
- [x] tag persistence-sdk 
- [x] remove pstake-native "replace" in go.mod after it is opensourced and tagged
- [x] cleanup

## 5. Limitations (optional)
- adding secondary auth module will be a little different

### CODE REVIEW

**Comment by puneet2019** [comment]
closes #119

File: app/app.go [hunk 1]
```diff
@@ -113,7 +128,6 @@ import (
 	tendermintdb "github.com/tendermint/tm-db"
 
 	appparams "github.com/persistenceOne/persistenceCore/v3/app/params"
```
Comment (Anmol1696) [inline app/app.go:130]: we need to upgrade all v3 imports to v4 as well no?

Comment (puneet2019) [reply app/app.go:130]: oo yea, will do

**Review by Anmol1696 — APPROVED** [review]
LGTM

**Pull request merged**